src/Termonad/Term.hs

```diff
@@ -425,64 +439,96 @@ setFocusOn tmStateAppWin vteTerm = do
-  maybeCurrDir <- getCWDFromFocusedTab currNote
-  termShellPid <- launchShell vteTerm maybeCurrDir
-  tmTerm <- newTMTerm vteTerm termShellPid
-
-  -- Create the container add the VTE term in it
-  scrolledWin <- createScrolledWin mvarTMState
-  containerAdd scrolledWin vteTerm
+  -- Launch a shell in a Terminal in a ScrolledWindow
+  let createTerm :: IO (ScrolledWindow, Terminal, TMTerm)
+      createTerm = do
+        scrolledWin <- createScrolledWin mvarTMState
+        vteTerm <- createAndInitVteTerm tmStateFontDesc (options tmStateConfig)
+        maybeCurrDir <- getCWDFromFocusedTab currNote
+        termShellPid <- launchShell vteTerm maybeCurrDir
+        tmTerm <- newTMTerm scrolledWin vteTerm termShellPid
+        containerAdd scrolledWin vteTerm
+        pure (scrolledWin, vteTerm, tmTerm)
+
+  -- Create the left and right terminals
+  (scrolledWinR, vteTermR, tmTermR) <- createTerm
+  (scrolledWinL, vteTermL, tmTermL) <- createTerm
+
+  -- Create the paned window container add the two terminals to it
+  paned <- Gtk.panedNew OrientationHorizontal
+  Gtk.panedSetWideHandle paned True
+  Gtk.panedPack1 paned scrolledWinL True True
+  Gtk.panedPack2 paned scrolledWinR True True
+  Gtk.widgetShowAll paned
 
   -- Create the GTK widget for the Notebook tab
   (tabLabelBox, tabLabel, tabCloseButton) <- createNotebookTabLabel
 
   -- Create notebook state
-  let notebookTab = createTMNotebookTab tabLabel scrolledWin tmTerm
+  let notebookTab = createTMNotebookTab tabLabel paned tmTermL tmTermR
 
   -- Add the new notebooktab to the notebook.
   addPage mvarTMState notebookTab tabLabelBox
 
   -- Setup the initial label for the notebook tab.  This needs to happen
   -- after we add the new page to the notebook, so that the page can get labelled
-  -- appropriately.
-  relabelTab (tmNotebook currNote) tabLabel scrolledWin vteTerm
+  -- appropriately. We use the left terminal's title because it is the left
+  -- terminal which initially has the focus.
+  relabelTab (tmNotebook currNote) tabLabel paned vteTermL
 
   -- Connect callbacks
   void $ onButtonClicked tabCloseButton $ termClose notebookTab mvarTMState
-  void $ onTerminalWindowTitleChanged vteTerm $ do
+  void $ onTerminalWindowTitleChanged vteTermL $ do
+    -- TODO: use the title of the focused pane
     TMState{tmStateNotebook} <- readMVar mvarTMState
     let notebook = tmNotebook tmStateNotebook
-    relabelTab notebook tabLabel scrolledWin vteTerm
-  void $ onWidgetKeyPressEvent vteTerm $ handleKeyPress mvarTMState
-  void $ onWidgetKeyPressEvent scrolledWin $ handleKeyPress mvarTMState
-  void $ onWidgetButtonPressEvent vteTerm $ handleMousePress vteTerm
-  void $ onTerminalChildExited vteTerm $ \_ -> termExit notebookTab mvarTMState
-
-  -- Put the keyboard focus on the term
-  setFocusOn tmStateAppWin vteTerm
+    relabelTab notebook tabLabel paned vteTermL
+  for_ @[Terminal] [vteTermL, vteTermR] $ \vteTerm -> do
```

Comment:
This looks like it might be a bad interaction between ClassyPrelude and `OverloadedLists`: https://github.com/cdepillabout/termonad/blob/36be8b08ede65000e9f5a7b566f33d9215f485c4/termonad.cabal#L122

I like your use of `TypeApplications` here.  It is very readable.